src/smt.rs

```diff
@@ -0,0 +1,95 @@
+use std::fmt::{Display, Formatter, Result};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SMTVariable {
+    pub name: String,
+}
+
+pub trait SMTSyntaxUnit {
+    fn as_smt(&self) -> String;
+}
+
+impl SMTSyntaxUnit for String {
+    fn as_smt(&self) -> String {
+        self.to_owned()
+    }
+}
+
+pub enum SMTLiteral {
+    Hex(String),
+}
+
+impl SMTLiteral {
+    pub fn hex_zero() -> Self {
+        Self::Hex("0".repeat(64))
+    }
+}
+
+pub enum SMTIdentifier {
+    Simple(String),
+    Indexed(String, usize), // for now supports only one index
```

Comment:
What would be other indices? This reminds me a bit of https://github.com/facebookincubator/smt2utils/blob/main/smt2parser/src/concrete.rs#L65
Wondering if we can get inspired by that module also

Reply:
there was no use case for multiple rn, but the [lib ref](https://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.6-r2017-07-18.pdf) stated that indexes can be repeated
<img width="600" alt="Screenshot 2022-09-12 at 11 55 38" src="https://user-images.githubusercontent.com/25429261/189613499-6558eadf-5b66-4e44-81ac-151f5d2d2849.png">
thanks for the lib link, i'll take a look, however, i think qualified identifiers are a bit different from regular ones

Reply:
Yep, agree.